### ISSUE TITLE
Logging

### ISSUE THREAD

**Issue by AngelIsSad** [Cyanic76/discord-modmail#19]
The log feature doesn't seem to work

**Comment by Cyanic76** [comment]
What do you mean?

**Comment by TK-Angel** [comment]
After you close a ticket or block someone, it doesnt get logged in the log channel

**Comment by Cyanic76** [comment]
Is there any error? Did you give enough perms?

**Comment by TK-Angel** [comment]
There are no errors and the Bot has Admin perms

**Comment by Cyanic76** [comment]
Can you please retry, just to make sure?

**Comment by Cyanic76** [comment]
Did you correctly set channel IDs in config.json?

**Comment by TK-Angel** [comment]
I did

**Comment by TK-Angel** [comment]
Everything works perfectly except the log

**Comment by Cyanic76** [comment]
Apparently the bot was not sending any embed because there was no declared embed or anything. Should be fixed in the next commit.

**Comment by Cyanic76** [comment]
There. Can you retry without setting config.showParticipants to true?

**Comment by TK-Angel** [comment]
You changed the code right?

**Comment by Cyanic76** [comment]
I did

**Comment by TK-Angel** [comment]
There is an error saying log is not defiend

**Comment by Cyanic76** [comment]
Refresh your code and try again?

**Comment by TK-Angel** [comment]
I did many times

**Comment by TK-Angel** [comment]
Error: Cannot read property 'forEach' of undefined

**Comment by TK-Angel** [comment]
TypeError: Cannot read property 'forEach' of undefined

**Comment by TK-Angel** [comment]
bot.js/line 273

**Comment by Cyanic76** [comment]
[There is literally nothing](https://github.com/Cyanic76/discord-modmail/blob/master/bot.js#L273) - did you put anything on line 273?

**Comment by TK-Angel** [comment]
I have this on there

participants.forEach(p => {

**Comment by TK-Angel** [comment]
Its line 237.... My bad

**Comment by TK-Angel** [comment]
Still the same error 
TypeError: Cannot read property 'forEach' of undefined

**Comment by TK-Angel** [comment]
Still doesn't work... Can i just send you my code?

**Comment by Cyanic76** [comment]
Sure, go ahead

**Comment by Cyanic76** [comment]
Can you remove lines 313 to 322?

**Comment by TK-Angel** [comment]
I can, but why?

**Comment by TK-Angel** [comment]
It works but it doesn't send the log to the log channel. It sends the closing message

https://cdn.discordapp.com/attachments/845989262317977610/847054800649977866/unknown.png

**Comment by TK-Angel** [comment]
Isn't it supposed to send the conversation between the mod and the other person or a link where you can read their convo?

**Comment by Cyanic76** [comment]
I'll soon add this feature.
Check your logs channel ID in your config.json file?

**Comment by TK-Angel** [comment]
The log is going to the correct channel but it is sending the closeing message

**Comment by Cyanic76** [comment]
the closing message that's being sent to the __thread channel__ is [here](https://github.com/Cyanic76/discord-modmail/blob/master/bot.js#L203) and the log is [here](https://github.com/Cyanic76/discord-modmail/blob/master/bot.js#L219). both are sent to the correct channels.

**Comment by TK-Angel** [comment]
It sends the line 220         

return client.users.cache.get(support.targetID).send(`Thanks for getting in touch with us. If you wish to open a new ticket, feel free to message me.\nYour ticket #${actualticket} has been closed.`)

**Comment by Cyanic76** [comment]
This is sent to the user, yeah

**Comment by TK-Angel** [comment]
Could you help me with the logs?

**Comment by TK-Angel** [comment]
?

**Comment by Cyanic76** [comment]
Yep, was a little busy - can you show me your code?

**Comment by TK-Angel** [comment]
The main.js?

**Comment by TK-Angel** [comment]
Bot.js: https://starb.in/2KDwHI.js

**Comment by Cyanic76** [comment]
[There you go, it's fixed](https://hastebin.com/memamixeda.http)

**Comment by TK-Angel** [comment]
There is 1 error with the mod role

**Comment by TK-Angel** [comment]
![Error](https://cdn.discordapp.com/attachments/823963920984965134/848119967551586314/unknown.png)

**Comment by TK-Angel** [comment]
The mod role id is correct

**Comment by TK-Angel** [comment]
moderator is not defined

**Comment by TK-Angel** [comment]
? It takes me to the top of the ticket

**Comment by Cyanic76** [comment]
Click [here](https://github.com/Cyanic76/discord-modmail/commit/e8ce25fe4587bffd644f09aa3f09a56faaba982c) instead

**Comment by TK-Angel** [comment]
Hey. It seems like the unblock command isn't working

**Comment by Cyanic76** [comment]
> Hey. It seems like the unblock command isn't working

What errors does it gives?

**Comment by TK-Angel** [comment]
It doesn't give any error in the terminal, it just sends the message `You can not use that` in the ticket channel even tho i have the mod role

**Comment by Cyanic76** [comment]
This is not a backend issue. The config.roles.mod value should be a **ROLE ID** and you should give that role to your mod team.

**Comment by AngelIsSad** [comment]
The value is a role ID and my Mod Team has that role

**Comment by Cyanic76** [comment]
It can only be a bad role ID or you not having it. Just checked the djs docs and nothing seems wrong.

**Comment by Cyanic76** [comment]
Hi @AngelIsSad, I successfully implemented your feature request using ubuntu's pastebin.

**Comment by AngelIsSad** [comment]
Thanks!
